agent/whois_ip_agent.py

```diff
@@ -54,19 +55,27 @@ def process(self, message: m.Message) -> None:
 
         else:
             host = message.data.get('host')
-            if not self.set_add('agent_whois_ip_asset', host):
-                logger.info('target %s was processed before, exiting', host)
+            mask = message.data.get('mask')
+
+            if mask is not None:
+                range_value = f'{host}/{mask}'
+            else:
+                range_value = host
+            if not self.set_add('agent_whois_ip_asset', range_value):
```

Comment:
If we process an IP with a mask, we must check all the higher denominators as covered as well for instance.
If we scan 8.8.8.0/24. we must check /23 with the right IP, /22, /21 ...
Those however should not use set_add because they should not add it, they should just check.